env/env_posix.cc

```diff
@@ -188,6 +188,15 @@ class PosixEnv : public Env {
       }
     }
     result->reset(new PosixSequentialFile(fname, file, fd, options));
+    {
+      struct stat buf;
+      int fstat_result = fstat(fd, &buf);
+      assert(fstat_result != -1);
```

Comment:
Can you explain in what cases it returns -1? If it is a valid case, rather than bug, I don't suggest we assert here, but just return the IOError.

Reply:
will remove the assert since it's a valid case.